src/main/scala/metadata.scala

```diff
@@ -0,0 +1,83 @@
+package suffuse
+
+/** AKey is an attribute key: a Metadata map key which carries
+ *  both the type of the corresponding value, and a value to
+ *  provide when there is no value in the map. This as opposed
+ *  to wrapping everything in sight in Option.
+ */
+final class AKey[A](description: String, empty: => A) {
```

Comment:
I wonder if this should be the same as `Key` in filesystem. Is this a uniform way to describe a `Key` (as in key-value) pair? A key-value pair implies some form of container in which there might not be a value associated with the key: thus the `empty` property.

I'm also wondering if it makes sense to have `empty` by name.